### PR TITLE
PXP-626: [CLI] Remove the tick-based canvas refresh

### PR DESCRIPTION
**Pull request by jk-gan**
## Summary

There is a performance issue on the logs panel, the ui will block the scrolling and input. With some investigations, **the bottleneck is came from data processing, not from the rendering**. We are processing far more data than we need in order to display on the screen.

For example, my screen only able to display 12 logs at a time, when we have 1000 logs in the memory, we are processing 1000 logs even though we only need 12 for rendering.

**So in conclusion, we don't have to remove the tick-based refresh. We just need to make sure we are using cpu to do meaningful stuffs.**

PS: The mouse clicking performance issue is fixed on #150 

Ticket: [PXP-626](https://mindvalley.atlassian.net/browse/PXP-626)

### Changed
- [x] Rework the scrolling, now it only process the data that are displayed on the screen

## Benchmark
Method: Scrolling through 1000 logs by holding down key

**Debug mode** means run the command using `cargo run -- tui`
**Release mode** means build the command using `cargo build --release` and run it

Before (Debug mode):
Screenshot 2023-10-12 at 2 40 23 PM
CPU **102.2%**

Before (Release mode):
Screenshot 2023-10-12 at 2 53 43 PM
CPU **55.4%**

After (Debug mode):
Screenshot 2023-10-12 at 2 36 46 PM
CPU **16.7%**

After (Release mode):
Screenshot 2023-10-12 at 2 58 44 PM
CPU **3.9%**

[PXP-626]: https://mindvalley.atlassian.net/browse/PXP-626?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/155#" title="Author self-approved">jk-gan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: cli/src/commands/tui/ui/logs.rs
Comment (mfauzaan) [inline cli/src/commands/tui/ui/logs.rs:74]: Whats the reason hiding the scrollbar?

**Review by mfauzaan — APPROVED** [review]
LGTM

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/155#" title="Author self-approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/155#pullrequestreview-1692087666" title="Approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: cli/src/commands/tui/ui/logs.rs
Comment (jk-gan) [reply cli/src/commands/tui/ui/logs.rs:74]: we are using the custom scrolling implementation, so the built-in scrollbar doesn't work well with that.

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/155#" title="Author self-approved">jk-gan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/155#" title="Author self-approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/155#pullrequestreview-1698626848" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
